amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/SynchronousGetRecordsRetrievalStrategy.java

```
@@ -45,9 +46,20 @@ public void shutdown() {
     public boolean isShutdown() {
         return false;
     }
-    
+
     @Override
-    public KinesisDataFetcher getDataFetcher() {
+    public Optional<DataFetcher> getDataFetcherOverride() {
+        return Optional.of(dataFetcher);
+    }
+
+    @Override
+    public DataFetcher dataFetcher() {
         return dataFetcher;
     }
+
+    @Override
+    public KinesisDataFetcher getDataFetcher() {
+        // returns null as dataFetcher() will now return the current dataFetcher
+        return null;
```

Let's throw UnsupportedOperationException if we expect this not to be called